src/plugins/Trader/apis/coinmarketcap/index.ts

```diff
@@ -0,0 +1,169 @@
+import CURRENCY_DATA from './currency.json'
+
+// proxy: https://web-api.coinmarketcap.com/v1
+const BASE_URL_v1 = 'https://coinmarketcap.provide.maskbook.com/v1'
+
+// proxy: https://web-api.coinmarketcap.com/v1.1
+const BASE_URL_v1_1 = 'https://coinmarketcap.provide.maskbook.com/v1'
+
+const WIDGET_BASE_URL = 'https://widgets.coinmarketcap.com/v2'
+
+export interface Status {
+    credit_count: number
+    elapsed: number
+    error_code: number
+    error_message: null | string
+    notice: null | string
+    timestamp: string
+}
+
+//#region get all currency
+export function getAllCurrenies() {
+    return CURRENCY_DATA
+}
+//#endregion
+
+//#region get all coins
+export interface Coin {
+    id: number
+    name: string
+    platform?: {
+        id: string
+        name: string
+        slug: string
+        symbol: string
+        token_address: string
+    }
+    rank: number
+    slug: string
+    status: 'active'
+    symbol: string
+}
+
+export async function getAllCoins() {
+    const response = await fetch(
+        `${BASE_URL_v1}/cryptocurrency/map?aux=status,platform&listing_status=active,untracked&sort=cmc_rank`,
```

Comment:
use `URLSearchParams`.

Reply:
No need for this one but I will update the others.